File: Archiver/app_code/clsIsolatedStorage.cs
// Imports System.Windows.Browser.HtmlPage
using global::System;
using System.Collections;
using global::System.IO;
using global::System.IO.IsolatedStorage;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using MODI;

namespace EcmArchiver
{
    public class clsIsolatedStorage
    {

        // Public bDoNotOverwriteExistingFile As Boolean = True
        // Public bOverwriteExistingFile As Boolean = False
        // Public bRestoreToOriginalDirectory As Boolean = False
        // Public bRestoreToMyDocuments As Boolean = False
        // Public bCreateOriginalDirIfMissing As Boolean = True

        public clsIsolatedStorage()
        {
            dirPersist = AppDir + @"\" + "EcmPersistantData";
            dirAttachInfo = AppDir + @"\" + "EcmAttachInfo";
            dirFormData = AppDir + @"\" + "EcmForm";
            dirTempData = AppDir + @"\" + "EcmTemp";
            dirLogData = AppDir + @"\" + "EcmLogs";
            dirSaveData = AppDir + @"\" + "EcmSavedData";

            // Dim store = IsolatedStorageFile.GetUserStoreForApplication()
            // isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User Or IsolatedStorageScope.Assembly Or IsolatedStorageScope.Domain, Nothing, Nothing)
            // Try
            // store.CreateDirectory(dirAttachInfo)
            // Catch ex As Exception
            // Console.WriteLine(ex.Message)
            // End Try

        }

        public string AppDir = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
        public string dirPersist;
        public string dirAttachInfo;
        public string dirFormData;
        public string dirTempData;
        public string dirLogData;
        public string dirSaveData;

        public void PersistDataInit(string tKey, string tData)
        {
            string FQN = Path.Combine(dirPersist, "Persist.dat");
            byte[] Buffer;
            string tgtLine = tKey + "|" + tData + Constants.vbCrLf;
            try
            {
                if (!Directory.Exists(dirPersist))
                {
                    Directory.CreateDirectory(dirPersist);
                }

                Buffer = StrToByteArray(tgtLine);
                var isoStream2 = new StreamWriter(FQN, false);
                isoStream2.WriteLine(tgtLine);
                isoStream2.Close();
            }
            catch (IsolatedStorageException ex)
            {
                MessageBox.Show("Error SaveFormData 700: saving data: " + ex.Message);
            }
        }

        public void PersistDataSave(string tKey, string tData)
        {

            // Dim dirFormData As String = "EcmTemp"
            string FQN = Path.Combine(dirPersist, "Persist.dat");
            // Dim isoStore As IsolatedStorageFile
            // isoStore = IsolatedStorageFile.GetUserStoreForApplication
            byte[] Buffer;
            string tgtLine = tKey + "|" + tData + Constants.vbCrLf;
            try
            {
                // Using store = IsolatedStorageFile.GetUserStoreForApplication()
                if (!Directory.Exists(dirPersist))
                {
                    Directory.CreateDirectory(dirPersist);
                }

                Buffer = StrToByteArray(tgtLine);
                var isoStream2 = new StreamWriter(FQN, true);
                isoStream2.Write(tgtLine);
                isoStream2.Close();
            }

            // End Using
            catch (IsolatedStorageException ex)
            {
                MessageBox.Show("Error SaveFormData 657: saving data: " + ex.Message);
            }
        }

        public string PersistDataRead(string tKey)
        {
            string FQN = Path.Combine(dirPersist, "Persist.dat");
            string ValName = "";
            string tgtVal = "";
            string RetVal = "";

            // Dim isoStore As IsolatedStorageFile
            // isoStore = IsolatedStorageFile.GetUserStoreForApplication
            // Using isoStore
            if (File.Exists(FQN))
            {
                using (var sr = new StreamReader(FQN))
                {
                    RetVal = "";
                    tgtVal = sr.ReadLine();
                    while (!sr.EndOfStream)
                    {
                        var A = tgtVal.Split('|');
                        if (A[0].Equals(tKey))
                        {
                            RetVal = A[1];
                            break;
                        }

                        tgtVal = sr.ReadLine();
                    }

                    sr.Close();
                }
            }

            // End Using

            return RetVal;
        }

        public string readIsoFile(string Filename)
        {
            // If Not File.Exists(Filename) Then
            // Return ""
            // End If
            string InputText = "";
            try
            {
                var isolatedStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Assembly, null, null);
                var isolatedStream = new IsolatedStorageFileStream(Filename, FileMode.Open, isolatedStore);
                var reader = new StreamReader(isolatedStream);
                InputText = reader.ReadToEnd();
                reader.Close();
            }
            catch (Exception ex)
            {
                if (!modGlobals.gRunUnattended)
                {
                    Console.WriteLine("Notice: Failed to READ data from isolated storage #122." + ex.Message);
                }
            }

            return InputText;
        }

        public void saveIsoFile(string FileName, string sLine)
        {
            try
            {
                var isolatedStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Assembly, null, null);
                var isoStream = new IsolatedStorageFileStream(FileName, FileMode.Append, FileAccess.Write, isolatedStore);
                var writer = new StreamWriter(isoStream);
                writer.WriteLine(sLine);
                writer.Close();
            }
            catch (Exception ex)
            {
                if (!modGlobals.gRunUnattended)
                {
                    MessageBox.Show("Notice: Failed to save data to isolated storage #121." + ex.Message);
                }
            }
        }

        public void saveIsoFileZeroize(string FileName, string sLine)
        {
            try
            {
                var isolatedStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Assembly, null, null);
                var isoStream = new IsolatedStorageFileStream(FileName, FileMode.Create, FileAccess.Write, isolatedStore);
                var writer = new StreamWriter(isoStream);
                writer.WriteLine(sLine);
                writer.Close();
            }
            catch (Exception ex)
            {
                if (!modGlobals.gRunUnattended)
                {
                    MessageBox.Show("Notice: Failed to save data to isolated storage #121." + ex.Message);
                }
            }
        }

        public static byte[] StrToByteArray(string str)
        {
            var encoding = new System.Text.UTF8Encoding();
            return encoding.GetBytes(str);
        } // StrToByteArray

        public bool isoDirExist()
        {
            // Using store = IsolatedStorageFile.GetUserStoreForApplication()
            using (var isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Assembly | IsolatedStorageScope.Domain, null, null))
            {
                foreach (string dName in isoStore.GetDirectoryNames(dirAttachInfo))
                {
                    if (dName.Equals(dirAttachInfo))
                    {
                        return true;
                    }
                }
            }

            return false;
        }

        public bool isoFileExist()
        {
            if (isoDirExist())
            {
                string FormDataFileName = "AttachInfo.dat";
                string FQN = Path.Combine(dirAttachInfo, FormDataFileName);
                using (var isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Assembly | IsolatedStorageScope.Domain, null, null))
                {
                    foreach (string fName in isoStore.GetFileNames(FQN))
                    {
                        if (fName.Equals("AttachInfo.dat"))
                        {
                            return true;
                        }
                    }
                }
            }

            return false;
        }

        public void ZeroizeAttachData()
        {
            string FormDataFileName = "AttachInfo.dat";
            string FQN = Path.Combine(dirAttachInfo, FormDataFileName);
            try
            {
                if (isoDirExist())
                {
                    using (var isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Assembly | IsolatedStorageScope.Domain, null, null))
                    {
                        isoStore.DeleteDirectory(dirAttachInfo);
                    }
                }
            }
            catch (IsolatedStorageException ex)
            {
                Console.WriteLine("Error SaveFormData 100: saving data: " + ex.Message);
            }
        }

        public void SaveAttachData(string CompanyID, string RepoID)
        {
            string FormDataFileName = "AttachInfo.dat";
            string FQN = Path.Combine(dirAttachInfo, FormDataFileName);
            string tgtLine = CompanyID + "|" + RepoID;
            IsolatedStorageFile isoStore;
            isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Assembly | IsolatedStorageScope.Domain, null, null);
            if (!isoDirExist())
            {
                isoStore.CreateDirectory(dirAttachInfo);
            }

            try
            {
                // Declare a new StreamWriter.
                StreamWriter writer = null;
                // Assign the writer to the store and the file TestStore.
                writer = new StreamWriter(new IsolatedStorageFileStream(FQN, FileMode.Create, isoStore));
                writer.WriteLine(tgtLine + Constants.vbCrLf);
                writer.Close();
                writer.Dispose();
                writer = null;
            }
            catch (IsolatedStorageException ex)
            {
                Console.WriteLine("Error SaveFormData 200: saving data: " + ex.Message);
            }

            isoStore.Dispose();
        }

        public bool ReadAttachData(ref string CompanyID, ref string RepoID)
        {
            if (!isoFileExist())
            {
                return false;
            }

            CompanyID = "";
            RepoID = "";
            IsolatedStorageFile isoStore;
            bool B = true;
            string FormDataFileName = "AttachInfo.dat";
            string FQN = Path.Combine(dirAttachInfo, FormDataFileName);
            string tgtLine = "";
            isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Assembly | IsolatedStorageScope.Domain, null, null);
            var reader = new StreamReader(new IsolatedStorageFileStream(FQN, FileMode.Open, isoStore));
            using (isoStore)
            {
                int fLen = 0;
                using (reader)
                {
                    try
                    {
                        tgtLine = reader.ReadLine();
                    }
                    catch (Exception ex)
                    {
                        return false;
                    }

                    var A = tgtLine.Split('|');
                    CompanyID = A[0];
                    RepoID = A[1];
                    reader.Close();
                    reader.Dispose();
                }
            }

            isoStore.Dispose();
            return B;
        }

        private bool FileExists(string FQN, IsolatedStorageFile storeFile)
        {
            string fileString = Path.GetFileName(FQN);
            string[] files;
            files = storeFile.GetFileNames(FQN);
            var fileList = new ArrayList(files);
            if (fileList.Count > 0)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}

File: Archiver/app_code/clsIsolatedStorage.vb
'Imports System.Windows.Browser.HtmlPage
Imports System
Imports System.IO
Imports System.IO.IsolatedStorage
Imports System.IO.IsolatedStorage.IsolatedStorageFile
Imports System.IO.IsolatedStorage.IsolatedStorageFileStream

Public Class clsIsolatedStorage

    Public AppDir As String = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)
    Public dirPersist As String = AppDir + "\" + "EcmPersistantData"
    Public dirAttachInfo As String = AppDir + "\" + "EcmAttachInfo"
    Public dirFormData As String = AppDir + "\" + "EcmForm"
    Public dirTempData As String = AppDir + "\" + "EcmTemp"
    Public dirLogData As String = AppDir + "\" + "EcmLogs"
    Public dirSaveData As String = AppDir + "\" + "EcmSavedData"

    'Public bDoNotOverwriteExistingFile As Boolean = True
    'Public bOverwriteExistingFile As Boolean = False
    'Public bRestoreToOriginalDirectory As Boolean = False
    'Public bRestoreToMyDocuments As Boolean = False
    'Public bCreateOriginalDirIfMissing As Boolean = True

    Sub New()

        'Dim store = IsolatedStorageFile.GetUserStoreForApplication()
        'isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User Or IsolatedStorageScope.Assembly Or IsolatedStorageScope.Domain, Nothing, Nothing)
        'Try
        '    store.CreateDirectory(dirAttachInfo)
        'Catch ex As Exception
        '    Console.WriteLine(ex.Message)
        'End Try

    End Sub

    Public Sub PersistDataInit(ByVal tKey As String, ByVal tData As String)

        Dim FQN As String = System.IO.Path.Combine(dirPersist, "Persist.dat")
        Dim Buffer() As Byte
        Dim tgtLine As String = tKey + "|" + tData + Environment.NewLine

        Try
            If Not Directory.Exists(dirPersist) Then
                Directory.CreateDirectory(dirPersist)
            End If

            Buffer = StrToByteArray(tgtLine)
            Dim isoStream2 As New StreamWriter(FQN, False)
            isoStream2.WriteLine(tgtLine)
            isoStream2.Close()
        Catch ex As IsolatedStorageException
            MessageBox.Show("Error SaveFormData 700: saving data: " + ex.Message)
        End Try
    End Sub

    Public Sub PersistDataSave(ByVal tKey As String, ByVal tData As String)

        'Dim dirFormData As String = "EcmTemp"
        Dim FQN As String = System.IO.Path.Combine(dirPersist, "Persist.dat")
        'Dim isoStore As IsolatedStorageFile
        'isoStore = IsolatedStorageFile.GetUserStoreForApplication
        Dim Buffer() As Byte
        Dim tgtLine As String = tKey + "|" + tData + Environment.NewLine
        Try
            'Using store = IsolatedStorageFile.GetUserStoreForApplication()
            If Not Directory.Exists(dirPersist) Then
                Directory.CreateDirectory(dirPersist)
            End If

            Buffer = StrToByteArray(tgtLine)
            Dim isoStream2 As New StreamWriter(FQN, True)
            isoStream2.Write(tgtLine)
            isoStream2.Close()

            'End Using
        Catch ex As IsolatedStorageException
            MessageBox.Show("Error SaveFormData 657: saving data: " + ex.Message)
        End Try
    End Sub

    Public Function PersistDataRead(ByVal tKey As String) As String

        Dim FQN As String = System.IO.Path.Combine(dirPersist, "Persist.dat")
        Dim ValName As String = ""
        Dim tgtVal As String = ""
        Dim RetVal As String = ""

        'Dim isoStore As IsolatedStorageFile
        'isoStore = IsolatedStorageFile.GetUserStoreForApplication
        'Using isoStore
        If File.Exists(FQN) Then
            Using sr As StreamReader = New StreamReader(FQN)
                RetVal = ""
                tgtVal = sr.ReadLine()
                Do While Not sr.EndOfStream
                    Dim A() As String = tgtVal.Split("|")
                    If A(0).Equals(tKey) Then
                        RetVal = A(1)
                        Exit Do
                    End If
                    tgtVal = sr.ReadLine()
                Loop
                sr.Close()
            End Using
        End If

        'End Using

        Return RetVal

    End Function

    Function readIsoFile(ByVal Filename As String) As String
        'If Not File.Exists(Filename) Then
        '    Return ""
        'End If
        Dim InputText As String = ""
        Try
            Dim isolatedStore As IsolatedStorageFile = IsolatedStorageFile.GetStore(IsolatedStorageScope.User Or IsolatedStorageScope.Assembly, Nothing, Nothing)
            Dim isolatedStream As New IsolatedStorageFileStream(Filename, FileMode.Open, isolatedStore)
            Dim reader As New StreamReader(isolatedStream)
            InputText = reader.ReadToEnd
            reader.Close()
        Catch ex As Exception
            If Not gRunUnattended Then
                Console.WriteLine("Notice: Failed to READ data from isolated storage #122." + ex.Message)
            End If
        End Try

        Return InputText
    End Function

    Public Sub saveIsoFile(ByVal FileName As String, ByVal sLine As String)
        Try
            Dim isolatedStore As IsolatedStorageFile = IsolatedStorageFile.GetStore(IsolatedStorageScope.User Or IsolatedStorageScope.Assembly, Nothing, Nothing)
            Dim isoStream As New IsolatedStorageFileStream(FileName, FileMode.Append, FileAccess.Write, isolatedStore)

            Dim writer As New StreamWriter(isoStream)
            writer.WriteLine(sLine)
            writer.Close()
        Catch ex As Exception
            If Not gRunUnattended Then
                MessageBox.Show("Notice: Failed to save data to isolated storage #121." + ex.Message)
            End If
        End Try
    End Sub
    Public Sub saveIsoFileZeroize(ByVal FileName As String, ByVal sLine As String)
        Try
            Dim isolatedStore As IsolatedStorageFile = IsolatedStorageFile.GetStore(IsolatedStorageScope.User Or IsolatedStorageScope.Assembly, Nothing, Nothing)
            Dim isoStream As New IsolatedStorageFileStream(FileName, FileMode.Create, FileAccess.Write, isolatedStore)

            Dim writer As New StreamWriter(isoStream)
            writer.WriteLine(sLine)
            writer.Close()

        Catch ex As Exception
            If Not gRunUnattended Then
                MessageBox.Show("Notice: Failed to save data to isolated storage #121." + ex.Message)
            End If
        End Try
    End Sub

    Public Shared Function StrToByteArray(ByVal str As String) As Byte()
        Dim encoding As New System.Text.UTF8Encoding()
        Return encoding.GetBytes(str)
    End Function 'StrToByteArray

    Public Function isoDirExist() As Boolean
        'Using store = IsolatedStorageFile.GetUserStoreForApplication()
        Using isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User Or IsolatedStorageScope.Assembly Or IsolatedStorageScope.Domain, Nothing, Nothing)
            For Each dName As String In isoStore.GetDirectoryNames(dirAttachInfo)
                If dName.Equals(dirAttachInfo) Then
                    Return True
                End If
            Next
        End Using
        Return False
    End Function
    Public Function isoFileExist() As Boolean
        If isoDirExist() Then
            Dim FormDataFileName As String = "AttachInfo.dat"
            Dim FQN As String = System.IO.Path.Combine(dirAttachInfo, FormDataFileName)
            Using isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User Or IsolatedStorageScope.Assembly Or IsolatedStorageScope.Domain, Nothing, Nothing)
                For Each fName As String In isoStore.GetFileNames(FQN)
                    If fName.Equals("AttachInfo.dat") Then
                        Return True
                    End If
                Next
            End Using
        End If
        Return False
    End Function

    Public Sub ZeroizeAttachData()
        Dim FormDataFileName As String = "AttachInfo.dat"
        Dim FQN As String = System.IO.Path.Combine(dirAttachInfo, FormDataFileName)

        Try
            If isoDirExist() Then
                Using isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User Or IsolatedStorageScope.Assembly Or IsolatedStorageScope.Domain, Nothing, Nothing)
                    isoStore.DeleteDirectory(dirAttachInfo)
                End Using
            End If
        Catch ex As IsolatedStorageException
            Console.WriteLine("Error SaveFormData 100: saving data: " + ex.Message)
        End Try
    End Sub

    Public Sub SaveAttachData(ByVal CompanyID As String, ByVal RepoID As String)

        Dim FormDataFileName As String = "AttachInfo.dat"
        Dim FQN As String = System.IO.Path.Combine(dirAttachInfo, FormDataFileName)
        Dim tgtLine As String = CompanyID + "|" + RepoID

        Dim isoStore As IsolatedStorageFile
        isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User Or IsolatedStorageScope.Assembly Or IsolatedStorageScope.Domain, Nothing, Nothing)

        If Not isoDirExist() Then
            isoStore.CreateDirectory(dirAttachInfo)
        End If

        Try
            ' Declare a new StreamWriter.
            Dim writer As StreamWriter = Nothing
            ' Assign the writer to the store and the file TestStore.
            writer = New StreamWriter(New IsolatedStorageFileStream(FQN, FileMode.Create, isoStore))
            writer.WriteLine(tgtLine + Environment.NewLine)
            writer.Close()
            writer.Dispose()
            writer = Nothing
        Catch ex As IsolatedStorageException
            Console.WriteLine("Error SaveFormData 200: saving data: " + ex.Message)
        End Try
        isoStore.Dispose()

    End Sub

    Public Function ReadAttachData(ByRef CompanyID As String, ByRef RepoID As String) As Boolean

        If Not Me.isoFileExist Then
            Return False
        End If

        CompanyID = ""
        RepoID = ""

        Dim isoStore As IsolatedStorageFile
        Dim B As Boolean = True
        Dim FormDataFileName As String = "AttachInfo.dat"
        Dim FQN As String = System.IO.Path.Combine(dirAttachInfo, FormDataFileName)
        Dim tgtLine As String = ""

        isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User Or IsolatedStorageScope.Assembly Or IsolatedStorageScope.Domain, Nothing, Nothing)
        Dim reader As New StreamReader(New IsolatedStorageFileStream(FQN, FileMode.Open, isoStore))
        Using isoStore
            Dim fLen As Integer = 0
            Using reader
                Try
                    tgtLine = reader.ReadLine()
                Catch ex As Exception
                    Return False
                End Try
                Dim A() As String = tgtLine.Split("|")
                CompanyID = A(0)
                RepoID = A(1)
                reader.Close()
                reader.Dispose()
            End Using
        End Using
        isoStore.Dispose()
        Return B

    End Function
    Private Function FileExists(FQN As String, storeFile As IsolatedStorageFile) As Boolean

        Dim fileString As String = Path.GetFileName(FQN)

        Dim files() As String
        files = storeFile.GetFileNames(FQN)

        Dim fileList As ArrayList = New ArrayList(files)

        If fileList.Count > 0 Then
            Return True
        Else
            Return False
        End If
      

    End Function


End Class


